### DISCUSSION TITLE
Naming convention for action-area objects on pages

### DISCUSSION THREAD

**Discussion by Arthurvdv** [StefanMaron/BusinessCentral.LinterCop#288] [New Rule]
With the introduction of the [Modern Action Bar](https://learn.microsoft.com/en-us/dynamics365-release-plan/2022wave2/smb/dynamics365-business-central/modern-action-bar) honestly I'm struggling a bit to have a consistent name convention these objects.

What could be the _best pratices_ on naming these?
* group
* action
* actionref

Example
```AL
actions
{
    area(processing)
    {
        group(Post) // PostGroup? PostGrp? Post_Group?
        {
            Caption = 'Post';

            action(Post)  // PostAction? PostAct? Post_Action?
            {
                Caption = 'Post';

                trigger OnAction()
                begin
                    Rec.Post();
                end;
            }
        }
    area(Promoted)
    {
        group(Post)  // PostPromotedGroup? PostPrmtdGrp? Post_Promoted_Group?
        {
            Caption = 'Post';

            actionref(Post; Post) { }  // PostPromoted? PostPrmtd? Post_Promoted? or PostActionref? PostActRef? Post_Actionref?
        }
    }
}
```

Does it make sense to have a rule to help having a consistent name? Are there maybe other area's where this could help and/or is needed?

**Comment by StefanMaron** [comment]
That might be a topic to raise here:
https://github.com/microsoft/alguidelines/discussions

**Reply by Arthurvdv** [reply]
Thanks, that a good idea!

I've created an topic (https://github.com/microsoft/alguidelines/discussions/223) and depending on the answer we can discuss if this has a place in the LinterCop.
